### PR TITLE
PWN-7606 - fixes for coingecko

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7606

## Description of Work

PWN-7606 - fixes for coingecko

### CODE REVIEW

**Review by gslevinkov — APPROVED** [review]

**Comment by gslevinkov** [comment]
don't forget to bump patch version

File: app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt [hunk 1]
```diff
@@ -36,19 +39,23 @@ class TokenPricesCoinGeckoRepository(
...
+                } catch (httpException: HttpException) {
+                    val errorCode = httpException.code()
+                    if (errorCode == 429 || errorCode == 403) {
+                        Timber.e(RequestRateLimitMet(httpException))
+                    }
+                    throw httpException
                 }
-                throw httpException
             }
+            priceCacheRepository.getPrices().entries.map { TokenPrice(tokenId = it.key, price = it.value.priceInUsd) }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt:59]: can you add ubit tests please?

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt:59]: > can you add unit tests please?
i do not even know what tests i can write here?
check that cache working? by timestamp?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt:59]: you can also add check that we fetch correct popular tokens and it's prices

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt:59]: we need to assign Test Guru on this test =) @gslevinkov 😈

**Pull request merged**